### PR TITLE
r/aws_evidently_launch

### PR DESCRIPTION
**Pull request by GlennChia**
### Description

`aws_evidently_launch` resource

### Relations

Relates #22624

### References

- [API_CreateLaunch](https://docs.aws.amazon.com/cloudwatchevidently/latest/APIReference/API_CreateLaunch.html)
- [API_GetLaunch](https://docs.aws.amazon.com/cloudwatchevidently/latest/APIReference/API_GetLaunch.html)
- [API_UpdateLaunch](https://docs.aws.amazon.com/cloudwatchevidently/latest/APIReference/API_UpdateLaunch.html)
- [API_DeleteLaunch](https://docs.aws.amazon.com/cloudwatchevidently/latest/APIReference/API_DeleteLaunch.html)

### Output from Acceptance Testing

```
$ make testacc TESTARGS='-run=TestAccEvidentlyLaunch' PKG=evidently
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/evidently/... -v -count 1 -parallel 20  -run=TestAccEvidentlyLaunch -timeout 180m
=== RUN   TestAccEvidentlyLaunch_basic
=== PAUSE TestAccEvidentlyLaunch_basic
=== RUN   TestAccEvidentlyLaunch_updateDescription
=== PAUSE TestAccEvidentlyLaunch_updateDescription
=== RUN   TestAccEvidentlyLaunch_updateGroups
=== PAUSE TestAccEvidentlyLaunch_updateGroups
=== RUN   TestAccEvidentlyLaunch_updateMetricMonitors
=== PAUSE TestAccEvidentlyLaunch_updateMetricMonitors
=== RUN   TestAccEvidentlyLaunch_updateRandomizationSalt
=== PAUSE TestAccEvidentlyLaunch_updateRandomizationSalt
=== RUN   TestAccEvidentlyLaunch_scheduledSplitsConfig_updateSteps
=== PAUSE TestAccEvidentlyLaunch_scheduledSplitsConfig_updateSteps
=== RUN   TestAccEvidentlyLaunch_scheduledSplitsConfig_steps_updateSegmentOverrides
=== PAUSE TestAccEvidentlyLaunch_scheduledSplitsConfig_steps_updateSegmentOverrides
=== RUN   TestAccEvidentlyLaunch_tags
=== PAUSE TestAccEvidentlyLaunch_tags
=== RUN   TestAccEvidentlyLaunch_disappears
=== PAUSE TestAccEvidentlyLaunch_disappears
=== CONT  TestAccEvidentlyLaunch_basic
=== CONT  TestAccEvidentlyLaunch_scheduledSplitsConfig_updateSteps
=== CONT  TestAccEvidentlyLaunch_disappears
=== CONT  TestAccEvidentlyLaunch_updateGroups
=== CONT  TestAccEvidentlyLaunch_tags
=== CONT  TestAccEvidentlyLaunch_scheduledSplitsConfig_steps_updateSegmentOverrides
=== CONT  TestAccEvidentlyLaunch_updateRandomizationSalt
=== CONT  TestAccEvidentlyLaunch_updateMetricMonitors
=== CONT  TestAccEvidentlyLaunch_updateDescription
--- PASS: TestAccEvidentlyLaunch_disappears (29.87s)
--- PASS: TestAccEvidentlyLaunch_basic (33.36s)
--- PASS: TestAccEvidentlyLaunch_updateDescription (50.54s)
--- PASS: TestAccEvidentlyLaunch_updateRandomizationSalt (51.83s)
--- PASS: TestAccEvidentlyLaunch_tags (65.68s)
--- PASS: TestAccEvidentlyLaunch_scheduledSplitsConfig_updateSteps (83.80s)
--- PASS: TestAccEvidentlyLaunch_updateMetricMonitors (85.43s)
--- PASS: TestAccEvidentlyLaunch_scheduledSplitsConfig_steps_updateSegmentOverrides (85.67s)
--- PASS: TestAccEvidentlyLaunch_updateGroups (86.47s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/evidently  86.585s

...
```

### CODE REVIEW

File: website/docs/r/evidently_launch.html.markdown [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+
+* `steps` - (Required) One or up to six blocks that define the traffic allocation percentages among the feature variations during each step of the launch. This also defines the start time of each step. [Detailed below](#steps).
+
+#### `steps`
+
+The `steps` block supports the following arguments:
+
+* `group_weights` - (Required) The traffic allocation percentages among the feature variations during one step of a launch. This is a set of key-value pairs. The keys are variation names. The values represent the percentage of traffic to allocate to that variation during this step. For more information, refer to the [AWS documentation for ScheduledSplitConfig groupWeights](https://docs.aws.amazon.com/cloudwatchevidently/latest/APIReference/API_ScheduledSplitConfig.html).
+* `segment_overrides` - (Required) One or up to six blocks that specify different traffic splits for one or more audience segments. A segment is a portion of your audience that share one or more characteristics. Examples could be Chrome browser users, users in Europe, or Firefox browser users in Europe who also fit other criteria that your application collects, such as age. [Detailed below](#segment_overrides).
```
Comment (drewmullen) [inline website/docs/r/evidently_launch.html.markdown:305]:
```suggestion
* `segment_overrides` - (Required) One or up to six blocks that specify different traffic splits for one or more audience segments. A segment is a portion of your audience that share one or more characteristics. Examples could be Chrome browser users, users in Europe, or Firefox browser users in Europe who also fit other criteria that your application collects, such as age. [Detailed below](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/evidently_launch#segment_overrides).
```

File: website/docs/r/evidently_launch.html.markdown [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+* `description` - (Optional) Specifies the description of the launch group.
+* `feature` - (Required) Specifies the name of the feature that the launch is using.
+* `name` - (Required) Specifies the name of the lahnch group.
+* `variation` - (Required) Specifies the feature variation to use for this launch group.
+
+### `metric_monitors`
+
+The `metric_monitors` block supports the following arguments:
+
+* `metric_definition` - (Required) A block that defines the metric. [Detailed below](#metric_definition).
```
Comment (drewmullen) [inline website/docs/r/evidently_launch.html.markdown:282]:
```suggestion
* `metric_definition` - (Required) A block that defines the metric. [Detailed below](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/evidently_launch#metric_definition).
```

File: website/docs/r/evidently_launch.html.markdown [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+
+The following arguments are supported:
+
+* `description` - (Optional) Specifies the description of the launch.
+* `groups` - (Required) One or up to five blocks that contain the feature and variations that are to be used for the launch. [Detailed below](#groups).
+* `metric_monitors` - (Optional) One or up to three blocks that define the metrics that will be used to monitor the launch performance. [Detailed below](#metric_monitors).
+* `name` - (Required) The name for the new launch. Minimum length of `1`. Maximum length of `127`.
+* `project` - (Required) The name or ARN of the project that is to contain the new launch.
+* `randomization_salt` - (Optional) When Evidently assigns a particular user session to a launch, it must use a randomization ID to determine which variation the user session is served. This randomization ID is a combination of the entity ID and randomizationSalt. If you omit randomizationSalt, Evidently uses the launch name as the randomizationSalt.
+* `scheduled_splits_config` - (Optional) A block that defines the traffic allocation percentages among the feature variations during each step of the launch. [Detailed below](#scheduled_splits_config).
```
Comment (drewmullen) [inline website/docs/r/evidently_launch.html.markdown:266]:
```suggestion
* `scheduled_splits_config` - (Optional) A block that defines the traffic allocation percentages among the feature variations during each step of the launch. [Detailed below](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/evidently_launch#scheduled_splits_config).
```

File: website/docs/r/evidently_launch.html.markdown [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `description` - (Optional) Specifies the description of the launch.
+* `groups` - (Required) One or up to five blocks that contain the feature and variations that are to be used for the launch. [Detailed below](#groups).
+* `metric_monitors` - (Optional) One or up to three blocks that define the metrics that will be used to monitor the launch performance. [Detailed below](#metric_monitors).
```
Comment (drewmullen) [inline website/docs/r/evidently_launch.html.markdown:262]:
```suggestion
* `metric_monitors` - (Optional) One or up to three blocks that define the metrics that will be used to monitor the launch performance. [Detailed below](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/evidently_launch#metric_monitors).
```

File: website/docs/r/evidently_launch.html.markdown [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `description` - (Optional) Specifies the description of the launch.
+* `groups` - (Required) One or up to five blocks that contain the feature and variations that are to be used for the launch. [Detailed below](#groups).
```
Comment (drewmullen) [inline website/docs/r/evidently_launch.html.markdown:261]:
```suggestion
* `groups` - (Required) One or up to five blocks that contain the feature and variations that are to be used for the launch. [Detailed below](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/evidently_launch#groups).
```

File: website/docs/r/evidently_launch.html.markdown [hunk 6]
```diff
@@ -0,0 +1,349 @@
...
+### `execution`
+
+The `execution` block supports the following attributes:
+
+* `ended_time` - The date and time that the launch ended.
+* `started_time` - The date and time that the launch started.
+
+## Import
+
+CloudWatch Evidently Launch can be imported using the launch `name` and `name` or `arn` of the hosting CloudWatch Evidently Project separated by a `:`, e.g.,
```
Comment (drewmullen) [inline website/docs/r/evidently_launch.html.markdown:345]: This is a little unclear, whats the diff b/w the launch name and name? perhaps add an example of each way to import

**Review by drewmullen — CHANGES_REQUESTED** [review]
1 small nitpick 

also, i believe internal markdown links should be proper URLs to work better with the registry although im not 100% sure on that

**Review by drewmullen — APPROVED** [review]
LGTM

File: website/docs/r/evidently_launch.html.markdown (hunk 4 above)
Comment (GlennChia) [reply website/docs/r/evidently_launch.html.markdown:262]: Following the Kendra docs https://github.com/hashicorp/terraform-provider-aws/blob/fcb0d1ed516f6036c145db8f1ea76709c2298382/website/docs/r/kendra_experience.html.markdown?plain=1#L49 full length URLs are not required for linking, pretty cool stuff

File: website/docs/r/evidently_launch.html.markdown (hunk 3 above)
Comment (GlennChia) [reply website/docs/r/evidently_launch.html.markdown:266]: Following the Kendra docs https://github.com/hashicorp/terraform-provider-aws/blob/fcb0d1ed516f6036c145db8f1ea76709c2298382/website/docs/r/kendra_experience.html.markdown?plain=1#L49 full length URLs are not required for linking, pretty cool stuff

File: website/docs/r/evidently_launch.html.markdown (hunk 2 above)
Comment (GlennChia) [reply website/docs/r/evidently_launch.html.markdown:282]: Following the Kendra docs https://github.com/hashicorp/terraform-provider-aws/blob/fcb0d1ed516f6036c145db8f1ea76709c2298382/website/docs/r/kendra_experience.html.markdown?plain=1#L49 full length URLs are not required for linking, pretty cool stuff

File: website/docs/r/evidently_launch.html.markdown (hunk 1 above)
Comment (GlennChia) [reply website/docs/r/evidently_launch.html.markdown:305]: Following the Kendra docs https://github.com/hashicorp/terraform-provider-aws/blob/fcb0d1ed516f6036c145db8f1ea76709c2298382/website/docs/r/kendra_experience.html.markdown?plain=1#L49 full length URLs are not required for linking, pretty cool stuff

**Review by jar-b — APPROVED** [review]
LGTM 🎉 

```console
$ make testacc TESTARGS='-run=TestAccEvidentlyLaunch' PKG=evidently
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/evidently/... -v -count 1 -parallel 20  -run=TestAccEvidentlyLaunch -timeout 180m
=== RUN   TestAccEvidentlyLaunch_basic
=== PAUSE TestAccEvidentlyLaunch_basic
=== RUN   TestAccEvidentlyLaunch_updateDescription
=== PAUSE TestAccEvidentlyLaunch_updateDescription
=== RUN   TestAccEvidentlyLaunch_updateGroups
=== PAUSE TestAccEvidentlyLaunch_updateGroups
=== RUN   TestAccEvidentlyLaunch_updateMetricMonitors
=== PAUSE TestAccEvidentlyLaunch_updateMetricMonitors
=== RUN   TestAccEvidentlyLaunch_updateRandomizationSalt
=== PAUSE TestAccEvidentlyLaunch_updateRandomizationSalt
=== RUN   TestAccEvidentlyLaunch_scheduledSplitsConfig_updateSteps
=== PAUSE TestAccEvidentlyLaunch_scheduledSplitsConfig_updateSteps
=== RUN   TestAccEvidentlyLaunch_scheduledSplitsConfig_steps_updateSegmentOverrides
=== PAUSE TestAccEvidentlyLaunch_scheduledSplitsConfig_steps_updateSegmentOverrides
=== RUN   TestAccEvidentlyLaunch_tags
=== PAUSE TestAccEvidentlyLaunch_tags
=== RUN   TestAccEvidentlyLaunch_disappears
=== PAUSE TestAccEvidentlyLaunch_disappears
=== CONT  TestAccEvidentlyLaunch_basic
=== CONT  TestAccEvidentlyLaunch_scheduledSplitsConfig_updateSteps
=== CONT  TestAccEvidentlyLaunch_tags
=== CONT  TestAccEvidentlyLaunch_updateMetricMonitors
=== CONT  TestAccEvidentlyLaunch_scheduledSplitsConfig_steps_updateSegmentOverrides
=== CONT  TestAccEvidentlyLaunch_updateGroups
=== CONT  TestAccEvidentlyLaunch_disappears
=== CONT  TestAccEvidentlyLaunch_updateRandomizationSalt
=== CONT  TestAccEvidentlyLaunch_updateDescription
--- PASS: TestAccEvidentlyLaunch_disappears (22.81s)
--- PASS: TestAccEvidentlyLaunch_basic (26.40s)
--- PASS: TestAccEvidentlyLaunch_updateDescription (33.86s)
--- PASS: TestAccEvidentlyLaunch_updateRandomizationSalt (33.88s)
--- PASS: TestAccEvidentlyLaunch_tags (45.76s)
--- PASS: TestAccEvidentlyLaunch_scheduledSplitsConfig_updateSteps (53.44s)
--- PASS: TestAccEvidentlyLaunch_scheduledSplitsConfig_steps_updateSegmentOverrides (54.54s)
--- PASS: TestAccEvidentlyLaunch_updateMetricMonitors (54.99s)
--- PASS: TestAccEvidentlyLaunch_updateGroups (58.32s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/evidently  61.462s
```

**Pull request merged**